Adding manage_bills.php script to scripts directory

rucarrol opened the pull request:
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [ X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/e9bbc1c3-8e19-48c2-b861-91c4938820f5) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/e9bbc1c3-8e19-48c2-b861-91c4938820f5) completed: **3 new issues, 4 updated code elements**

laf commented:
Thanks for this.

I've made a few changes with a git diff if you want to take a look: https://p.libren.ms/view/397b1866

Any reason you don't check hostname rather than sysName?

Can you reverse the default of `-r` as it could lead to confusion if it wipes out existing bills.

rucarrol commented:
> I've made a few changes with a git diff if you want to take a look: https://p.libren.ms/view/397b1866

Nice, thank you! 

> Any reason you don't check hostname rather than sysName?

Mainly as the set of devices I'm working on have a `hostname` which is not the same as `sysName`. I have no real opinion either way, but my reasoning was mostly that sysName would be what the device things it is, and `hostname` would be what DNS thinks it is (Which could change). Happy to pick `hostname` if that's the established way of doing it. 

> Can you reverse the default of -r as it could lead to confusion if it wipes out existing bills.

Sure - should I change the name to flush, so it becomes a bit more obvious?

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/9c1b7aee-d60a-4930-ae8c-923b5c7d2da4) was created.

laf commented:
> Mainly as the set of devices I'm working on have a hostname which is not the same as sysName. I have no real opinion either way, but my reasoning was mostly that sysName would be what the device things it is, and hostname would be what DNS thinks it is (Which could change). Happy to pick hostname if that's the established way of doing it.

Hostname would be better as it's unique, sysName isn't always depending on the devices :(

You could have two flags so people can pass either hostname or sysname?

> Sure - should I change the name to flush, so it becomes a bit more obvious?

Flush seems good.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/9c1b7aee-d60a-4930-ae8c-923b5c7d2da4) completed: **3 new issues, 4 updated code elements**

rucarrol commented:
> You could have two flags so people can pass either hostname or sysname?

Will do, tomorrow. 

> Flush seems good.

Please see [adfa1e965ac3e928b69411c6ae7f15ebf671957ahttps://github.com/librenms/librenms/pull/7633/commits/adfa1e965ac3e928b69411c6ae7f15ebf671957a).

laf commented:
Looks good to me :)

laf commented:
@rucarrol Any joy on the last few changes?

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/81e1cf57-ff88-42d6-adc3-0e2ea6dab9e7) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/81e1cf57-ff88-42d6-adc3-0e2ea6dab9e7) completed: **3 new issues, 4 updated code elements**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/2c9af8d0-faae-4278-93f7-2f6f8fa94a10) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/2c9af8d0-faae-4278-93f7-2f6f8fa94a10) completed: **3 new issues, 4 updated code elements**

rucarrol commented:
Hey @laf , just committed the diff :)

laf commented on scripts/manage_bills.php:
```diff
@@ -0,0 +1,144 @@
+$bill_name = str_replace('*', '%', mres($options['l']));
+$intf_glob = str_replace('*', '%', mres($options['c']));
+
+if (empty($bill_name)) {
```
This should also check that no -h and -s have been specified so that running the script with no args will give you this help.

laf commented on scripts/manage_bills.php:
```diff
@@ -0,0 +1,144 @@
+
+if (empty($bill_name)) {
+    echo "Usage:\n";
+    echo "-l <bill name glob>   Bill name to match\n";
```
`-b` instead maybe for bill name?

laf commented on scripts/manage_bills.php:
```diff
@@ -0,0 +1,144 @@
+    echo "-l <bill name glob>   Bill name to match\n";
+    echo "-s <sysName glob>     sysName to match (Cannot be used with -h)\n";
+    echo "-h <hostname glob>    Hostname to match (Cannot be used with -s)\n";
+    echo "-c <Interface description glob>   Interface description to match\n";
```
`-i` for interface description maybe?

laf commented on scripts/manage_bills.php:
```diff
@@ -0,0 +1,144 @@
+if ($bill_name == 'all') {
+    $bill_name = '%';
+}
+if ($intf_glob == 'all') {
```
Seeing as you can specify all interfaces I think we should bail the script if no `-c` (or `-i`) is set otherwise interfaces are added again

laf commented on scripts/manage_bills.php:
```diff
@@ -0,0 +1,144 @@
+}
+
+
+function create_bill($devs, $intf_glob, $id)
```
I'd rename this to `add_ports` so it clarifies what it does.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/e6982d22-0144-4b42-9e47-d284b71e2db7) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/e6982d22-0144-4b42-9e47-d284b71e2db7) completed: **1 new issues**

rucarrol commented:
Hey @laf 

Code review was much appreciated, thank you for taking the time to do this.

I changed the flow logic around a bit for detecting -h/-s, so that it'll print warnings, then help. 

Not setting -i will also cause the script to bail out. 

I also renamed the create_bill function to add_ports_to_bill (So I would not clobber the add_ports functioning already existing). 

Cheers + thanks against for assisting.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/77b22dfe-3aeb-4189-8bc6-376f9d85900b) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/77b22dfe-3aeb-4189-8bc6-376f9d85900b) completed: **1 new issues**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/143d009a-5ba0-497a-8685-ef1974cac01d) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/6eabbc2e-bb5a-4a32-9988-0c84236eb6b5) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/143d009a-5ba0-497a-8685-ef1974cac01d) completed: **3 new issues, 4 updated code elements**

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/6eabbc2e-bb5a-4a32-9988-0c84236eb6b5) completed: **3 new issues, 4 updated code elements**
